### ISSUE TITLE
Misleading preprocessor error message with character literals, especially within dialogs

### ISSUE THREAD

**Issue by fernewelten** [adventuregamestudio/ags#2609]
**Describe the bug**

I hit on a big snag in a game that has > 20 dialogs: When I compiled (F7), the system complained `Unterminated string` in file `__DialogScripts.asc`, line 1384. I couldn't navigate to the error by double-clicking the message, so I was tasked with finding the rogue line in about 1500 lines of dialog that were distributed in 20 dialog "files". This was time consuming. (This aspect is reported in #2389.)

When I found the rogue line at long last, it turned out to be the following:
```
    EGO: I've died!
```
Note the _indentation_, which turns the line into an AGS statement instead of a dialog line. The system found the `'` in ‘I've’ and complained about an `Unterminated string`. 

It hadn't occurred to me at all that this might even be the problem: I had been looking for a missing `"`, not for a missing `'` because the error message complained about a ‘string’. 
So at the end of the day, I lost a lot of time and nerves in locating and hunting down the wrong problem in a big bunch of big "files".

How can we better help AGS programmers that are stuck in this kind of situation? The cryptic message `Unterminated string` isn't it IMHO.

**AGS Version**
AGS 4 branch, commit dfc68e28
Also happens within AGS 3.6.1

As far as I can see, the error message comes out of file preprocessor.cpp of the classic compiler, line 248 and/or file Preprocessor.cs of the Editor, line 391

**To Reproduce**
* In a new AGS game that has the character `cEgo`
* Create a new dialog
* As the text of the dialog, enter:
```
@S
    EGO: I've
```
* Compile (F7)

**Comment by ericoporto** [comment]
I think perhaps, adding line number as parameter in `ConvertDialogScriptLine`

https://github.com/adventuregamestudio/ags/blob/e0854823da9ce01215858db7ecc3e0e32b3293c3/Editor/AGS.Editor/Utils/DialogScriptConverter.cs#L107

and then it could do a check that if a line starts with `[ \t]+[a-zA-Z0-9]+:` , meaning, any number of spaces, followed by any number of valid keyword characters followed by a `:` character, it should then throw a warning that "this line is probably not a script".

**Comment by ivan-mogilko** [comment]
#2615 is merged, and it changed the preprocessor error to 
> Unterminated string: " is missing
or
> Unterminated string: ' is missing

Is this enough to close this?
The above suggestion by @ericoporto may be also implemented, it may catch certain kinds of mistakes earlier.

**Comment by ivan-mogilko** [comment]
> and then it could do a check that if a line starts with `[ \t]+[a-zA-Z0-9]+:` , meaning, any number of spaces, followed by any number of valid keyword characters followed by a `:` character, it should then throw a warning that "this line is probably not a script".

So, I tried to do this, but unfortunately this pattern won't be reliable after we have named parameters in function call.
Because in theory one may have following script:

```
EGO: dialog line

    int x = func(
        a: 10, b: 20);
```

Then such search pattern will also find `a:`.
We may also get more of the same syntax if proposed array and struct initializer lists are supported.

I don't have any other immediate ideas on how to catch such mistakes during dialog script conversion.
I think I'll close this issue now, since the reported problem has been fixed.
